nexus/types/src/deployment.rs

```diff
@@ -1000,6 +1043,193 @@ impl fmt::Display for BlueprintZoneImageVersion {
     }
 }
 
+#[derive(
+    Clone, Debug, Eq, PartialEq, JsonSchema, Deserialize, Serialize, Diffable,
+)]
+pub struct PendingMgsUpdates {
+    // Only one outstanding MGS-managed update is allowed for a given baseboard.
+    by_baseboard: BTreeMap<Arc<BaseboardId>, PendingMgsUpdate>,
+}
+
+impl PendingMgsUpdates {
+    pub fn new() -> PendingMgsUpdates {
+        PendingMgsUpdates { by_baseboard: BTreeMap::new() }
+    }
+
+    pub fn iter(
+        &self,
+    ) -> impl Iterator<Item = (&Arc<BaseboardId>, &PendingMgsUpdate)> {
+        self.into_iter()
+    }
+
+    pub fn len(&self) -> usize {
+        self.by_baseboard.len()
+    }
+
+    pub fn is_empty(&self) -> bool {
+        self.by_baseboard.is_empty()
+    }
+
+    pub fn contains_key(&self, key: &Arc<BaseboardId>) -> bool {
+        self.by_baseboard.contains_key(key)
+    }
+
+    pub fn get(&self, baseboard_id: &BaseboardId) -> Option<&PendingMgsUpdate> {
+        self.by_baseboard.get(baseboard_id)
+    }
+
+    pub fn values(&self) -> impl Iterator<Item = &PendingMgsUpdate> {
+        self.by_baseboard.values()
+    }
+
+    pub fn remove(
+        &mut self,
+        baseboard_id: &BaseboardId,
+    ) -> Option<PendingMgsUpdate> {
+        self.by_baseboard.remove(baseboard_id)
+    }
+
+    pub fn add_or_replace(
```

Comment:
Nit but I think it'd be fine to name this `insert` since "insert returns an Option to indicate whether it was an add or replace" is the normal API for BTreeMap/HashMap.

Reply:
Done in 244c42b4b2c750f73e355a614b00e64e3f126874.